content/docs/08-troubleshooting/01-migration-guide/04-migration-guide-4-0.mdx

```diff
@@ -67,3 +67,13 @@ The `name` property from the `Message` type has been removed.
 
 The `streamMode` parameter has been removed from the `useChat` and `useCompletion` hooks.
 Please use the `streamProtocol` parameter instead.
+
+### `useChat`: Replaced roundtrip options with `maxSteps`
+
+The following options have been removed from the `useChat` hook:
+
+- `experimental_maxAutomaticRoundtrips`
+- `maxAutomaticRoundtrips`
+- `maxToolRoundtrips`
+
+Please use the `maxSteps` parameter instead.
```

Comment:
```suggestion
Please use the [`maxSteps`](/docs/ai-sdk-core/tools-and-tool-calling#multi-step-calls) parameter instead. Note: `maxStep` is equal to `maxAutomaticRoundtrips` + 1.
```

Reply:
Thought it might be nice to have the note. But equally happy to just link out to the main page.